source/docs/magento2/add-new-attribute.md

```diff
@@ -0,0 +1,187 @@
+---
+id: add-new-attribute
+title: Add a new attribute
+---
+
+Let's say we want to add a new element to display on a product's page (for instance, a product's rate). You will have to:
+
+1. Add the attribute's actual value to your database.
+2. Create a new module.
+3. Define the schema, implement the resolver and display it on screen.
+
+## Add a value to your database
+
+The process to do so may vary with the backend software you are using. For Magento 2 (used in this example), you can refer to [this](https://www.fastcomet.com/tutorials/magento2/product-attributes).
+
+## Create a new GraphQL module
+
+We want to display the `rate` of a product on the product page. To keep it simple, it will only consist in displaying a number, the value can be manually edited by the merchant. This feature will be stored in a new module called `mymodule` ; you could directly edit the files at `node_modules/front-commerce/path/to/something`, but this is not recommended since running `npm install` might erase any modification.
+Learn more about Front-Commerce's folder structure [here](/docs/essentials/extend-the-theme.html).
+
+```bash
+mkdir mymodule
+cd mymodule
+```
+
+In this example, the files we are going to override are located at `node_modules/front-commerce/src/web/theme/modules/ProductView/Synthesis/`, so will create in our `mymodule` folder the following directories. More details [here](/docs/essentials/extend-the-graphql-schema.html).
+
+```bash
+mkdir -p server/modules/acme
+```
+
+(Note that the `acme` name has been chosen arbitrarily.)
+
+## Add the module to Front-Commerce
+
+Now we will add the module we just made to Front-Commerce. Edit the `.front-commerce.js` file (root folder) as shown below to do so.
+
+```diff
+module.exports = {
+  name: "Front Commerce Skeleton",
+  url: "http://localhost:4000",
+- modules: ["./src"],
++ modules: ["./mymodule", "./src"],
+  serverModules: [
+    { name: "FrontCommerce", path: "server/modules/front-commerce" },
+-   { name: "Magento2", path: "server/modules/magento2" }
++   { name: "Magento2", path: "server/modules/magento2" },
++   { name: "Acme", path: "./mymodule/server/modules/acme" }
+  ]
+};
+```
+
+### Expand the Product definition
+
+Create a file named `schema.gql` and type in the following code. It means that we are expanding the definition of a product: it can now be rated.
+
+```js
+// mymodule/server/modules/acme/schema.gql
+extend type Product {
+  rate: Float
+}
+```
+
+### Implement the resolver
+
+We must now define what Front-Commerce should fetch when `rate` is requested in a GraphQL query; in other words we must write the code that will **resolve** the queries for the rates.
+
+Create a file named `resolvers.js` and type in the following code. It implements the resolver.
+
+```js
+// mymodule/server/modules/acme/resolvers.js
+export default {
+  Product: {
+    rate: parent => {
+      return parseFloat(parent.rate);
```

Comment:
Why did you use the `parent` directly?

Maybe you could add a paragraph explaining that by default all the attributes are available in a product. It's just that they are not exposed on the GraphQL schema yet.

Reply:
Is it better like this ? I hadn't thought thought of destructuring before.

```js
// mymodule/server/modules/ratings/resolvers.js
export default {
  Product: {
    rate: {rate} => parseFloat(rate);
  }
};
```

By default all the attributes are available in a product. It's just that they are not exposed on the GraphQL schema yet.